OpenScienceJournal/app/src/main/java/com/google/android/apps/forscience/whistlepunk/opensource/accounts/GoogleAccountsProvider.java

```diff
@@ -0,0 +1,123 @@
+package com.google.android.apps.forscience.whistlepunk.opensource.accounts;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+import androidx.fragment.app.Fragment;
+import com.google.android.apps.forscience.whistlepunk.ActivityWithNavigationView;
+import com.google.android.apps.forscience.whistlepunk.accounts.AbstractAccountsProvider;
+import com.google.android.apps.forscience.whistlepunk.accounts.AppAccount;
+import com.google.android.apps.forscience.whistlepunk.accounts.NonSignedInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.Task;
+import io.reactivex.Single;
+
+/** An accounts provider which is backed by Google Accounts. */
+@SuppressWarnings("RedundantIfStatement")
+public final class GoogleAccountsProvider extends AbstractAccountsProvider {
+  private static final String TAG = "GoogleAccountsProvider";
+
+  GoogleSignInClient googleSignInClient;
+  Context context;
+
+  public GoogleAccountsProvider(Context context) {
+    super(context);
+    this.context = context;
+    setShowSignInActivityIfNotSignedIn(false);
+    GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+        .requestEmail()
+        .build();
+    googleSignInClient = GoogleSignIn.getClient(context, gso);
+
+    GoogleSignInAccount googleSignInAccount = GoogleSignIn.getLastSignedInAccount(context);
+    if (googleSignInAccount != null) {
+      signInCurrentAccount(googleSignInAccount);
+    }
+  }
+
+  @Override
+  protected Single<PermissionStatus> isAccountPermitted(Activity activity, AppAccount appAccount) {
+    return Single.just(PermissionStatus.PERMITTED);
+  }
+
+
+  @Override
+  protected void afterSetCurrentAccount(AppAccount currentAccount) {
+    super.afterSetCurrentAccount(currentAccount);
+  }
+
+  @Override
+  public boolean supportSignedInAccount() {
+    return true;
+  }
+
+  @Override
+  public boolean isSignedIn() {
+    GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(context);
+    return account != null;
+  }
+
+  @Override
+  public int getAccountCount() {
+    GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(context);
+    return account == null ? 0 : 1;
+  }
+
+  @Override
+  public void installAccountSwitcher(ActivityWithNavigationView activity) {
+
+  }
```

Comment:
Liz probably knows this code better than me but does this being blank mean we can only sign into one account and not switch between them? If so, is that infeasible or should we file a bug to address that in the future?

Reply:
Correct. We can't switch between accounts. You sign in, then sign out, then sign in as someone else, if you want.

However, this method is actually the part of the code where we added the component to our app. We no longer do that at all, just intent off to a system dialog. So I don't think we'd have to do anything here anyway.